app/ibc_proposal_handler.go

```diff
@@ -1,28 +0,0 @@
-package app
-
-import (
-	"cosmossdk.io/errors"
-	sdk "github.com/cosmos/cosmos-sdk/types"
-	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
-	govtypes "github.com/cosmos/cosmos-sdk/x/gov/types/v1beta1"
-
-	"github.com/cosmos/ibc-go/v6/modules/core/02-client/keeper"
-	"github.com/cosmos/ibc-go/v6/modules/core/02-client/types"
-)
-
-// NewClientProposalHandler defines the 02-client proposal handler. It disables the
-// UpgradeProposalType. Handling of updating the IBC Client will be done in v2 of the
-// app.
-func NewClientProposalHandler(k keeper.Keeper) govtypes.Handler {
-	return func(ctx sdk.Context, content govtypes.Content) error {
-		switch c := content.(type) {
-		case *types.ClientUpdateProposal:
-			return k.ClientUpdateProposal(ctx, c)
-		case *types.UpgradeProposal:
-			return errors.Wrap(sdkerrors.ErrInvalidRequest, "ibc upgrade proposal not supported")
-
-		default:
-			return errors.Wrapf(sdkerrors.ErrUnknownRequest, "unrecognized ibc proposal content type: %T", c)
-		}
-	}
-}
```

Comment:
[question]

@cmwaters did we ever resolve this in v2? is this ok to remove?

Reply:
we removed this on the 50 branch too. these proposal handler msgs are v1beta1 version of x/gov. V1 x/gov supports arbitrary execution of sdk.Msg interface so we can favour that. 

We can disallow the software upgrade ibc proposal using x/circuit (similarly to the x/upgrade msgs)